### PR TITLE
Don't mark libc++ symbols as visibility:default

### PR DESCRIPTION
**Pull request by sbc100**
This means they won't get exports by default by lld.  This
is really workaround until we have a better way to implement
EMSCRIPTEN_KEEPALIVE.

This reduces the number of exports in C++ hello world from 1922
to 222.

### CODE REVIEW

**Comment by sbc100** [comment]
Partial fix for #7383

File: system/include/libcxx/__config [hunk 1]
```diff
@@ -39,6 +39,11 @@
 #define _LIBCPP_ABI_VERSION 1
 #endif
 
+#ifdef defined(__EMSCRIPTEN__) // XXX EMSCRIPTEN: Added to avoid making symbols as visibility(__default__) which for the wasm backend will export symbols to JS
+// FIXME(https://github.com/kripken/emscripten/issues/7383)
+#define _LIBCPP_DISABLE_VISIBILITY_ANNOTATIONS
```
Comment (kripken) [inline system/include/libcxx/__config]: Minor detail on how to implement this: Another option is to add this define in `tools/system_libs.py`. In general I think fewer changes to libc++ code is better, so I'd lean towards that, but maybe I'm missing something?

Comment (waywardmonkeys) [reply system/include/libcxx/__config]: I agree with @kripken here.

**Comment by sbc100** [comment]
Should be better now..

**Pull request merged**